language_service/src/rename.rs

```diff
@@ -0,0 +1,375 @@
+                if self.is_prepare {
+                    self.prepare = Some((ast_name.span, ast_name.name.to_string()));
+                } else {
+                    let def_span = match &def.kind {
+                        hir::ItemKind::Callable(decl) => decl.name.span,
+                        hir::ItemKind::Namespace(name, _) | hir::ItemKind::Ty(name, _) => name.span,
+                    };
+                    let mut rename = ItemRename {
+                        item_id,
+                        compilation: self.compilation,
+                        locations: vec![],
+                    };
+                    rename.visit_package(&self.compilation.unit.ast.package);
+                    rename.locations.push(def_span);
+                    self.locations = rename.locations;
+                }
+            }
+        }
+    }
+
+    fn get_spans_for_field_rename(&mut self, item_id: &hir::ItemId, ast_name: &ast::Ident) {
+        // Only rename items that are part of the local package
+        if item_id.package.is_none() {
+            if let Some(def) = self.compilation.unit.package.items.get(item_id.item) {
+                if let hir::ItemKind::Ty(_, udt) = &def.kind {
+                    if let Some(ty_field) = udt.find_field_by_name(&ast_name.name) {
+                        if self.is_prepare {
+                            self.prepare = Some((ast_name.span, ast_name.name.to_string()));
+                        } else {
+                            let def_span = ty_field
+                                .name_span
+                                .expect("field found via name should have a name");
+                            let mut rename = FieldRename {
+                                item_id,
+                                field_name: ast_name.name.clone(),
+                                compilation: self.compilation,
+                                locations: vec![],
+                            };
+                            rename.visit_package(&self.compilation.unit.ast.package);
+                            rename.locations.push(def_span);
+                            self.locations = rename.locations;
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    fn get_spans_for_local_rename(&mut self, node_id: ast::NodeId, ast_name: &ast::Ident) {
+        if let Some(curr) = self.current_callable {
+            if self.is_prepare {
+                self.prepare = Some((ast_name.span, ast_name.name.to_string()));
+            } else {
+                let mut rename = LocalRename {
+                    node_id,
+                    compilation: self.compilation,
+                    locations: vec![],
+                };
+                rename.visit_callable_decl(curr);
+                self.locations = rename.locations;
+            }
+        }
+    }
+}
+
+fn span_contains(span: Span, offset: u32) -> bool {
```

Comment:
Don't we already have this in qsc_utils.rs?

Reply:
This one is slightly different in that it includes the `span.hi` location. I've added a comment clarifying this distinction to the code.

Reply:
Discussed offline: decided to add a `source_touches` function to the utils for the difference cases involved.